R/encoloc.R

```diff
@@ -68,6 +68,123 @@ xqtl_enrichment_wrapper <- function(xqtl_files, gwas_files,
 } 
 
 
+
+#' Functions for  Colocalization Analysis Wrapper
+
+# Function to filter and order colocalization results
+filter_and_order_coloc_results <- function(coloc_results_fil) {
+  # Ensure the input has more than one column
+  if (ncol(coloc_results_fil) <= 1) {
+    stop("Insufficient number of columns in colocalization results")
+  }
+
+  cs_num <- ncol(coloc_results_fil) - 1
+  ordered_results <- list()
+
+  for (n in 1:cs_num) {
+    # Selecting relevant columns and ordering
+    tmp_coloc_results_fil <- coloc_results_fil[, c(1, n+1)] %>% .[order(.[, 2], decreasing = TRUE), ]
+    ordered_results[[n]] <- tmp_coloc_results_fil
+  }
+
+  return(ordered_results)
+}
+
+# Function to calculate cumulative sum
+calculate_cumsum <- function(coloc_results) {
+  cumsum(coloc_results[, 2])
+}
+
+# Function to load and extract LD matrix
+load_and_extract_ld_matrix <- function(ld_meta_file_path, analysis_region, variants) {
+  # This is a placeholder for loading LD matrix, adjust as per your actual function
+  ld_ref <- load_LD_matrix(LD_meta_file_path = ld_meta_file_path, region = analysis_region)
+  ext_ld <- ld_ref[[2]][variants, variants]
+  ext_ld
+}
+
+#' @importFrom susieR get_purity
+# Function to calculate purity
+calculate_purity <- function(variants, ext_ld, squared) {
+  # This is a placeholder for calculating purity, adjust as per your actual function
+  purity <- matrix(get_purity(variants, Xcorr = ext_ld, squared), 1, 3)
+  purity
+}
+
+# Main processing function
+process_coloc_results <- function(coloc_result, LD_meta_file_path,analysis_script_obj, PPH4_thres = 0.8, coloc_pip_thres = 0.95, squared = FALSE, min_abs_corr = 0.5, null_index = 0, coloc_index = "PP.H4.abf", analysis_region) {
```

Comment:
```r
      fitted[[r]]$region_info = list(region=parse_region("${_meta_info[0]}"), name="${_meta_info[2]}")
```

I add the above to https://github.com/cumc/xqtl-pipeline/commit/cd5398fed292fb87ffe82d33314f647d1c7aab38 so you don't need to parse from script loaded

Reply:
Also regarding purity -- in my branch of susieR I also use median abs corr, https://github.com/gaow/susieR/blob/master/R/susie_utils.R#L328 that you can also consider?

Reply:
Hi @gaow , in your version I found that you don't consider squared anymore, do we still keep that choice? And also you did some modifications on get purity. If we decide to use the new version, I will replace this whole part. Thanks!

Reply:
@rfeng2023 i don't think we ever considered squared correlation. It is always abs correlation either the min or median. You can use my new version to be consistent with our susieR analysis.

Reply:
@gaow okay. will do